src/Context/UserInfo.tsx

```diff
@@ -0,0 +1,115 @@
+import { createContext, useContext, useReducer } from "react";
+import { ICar, IUserContext } from "../interfaces/User";
+
+const initialState = {
+  name: "Gerardo",
+  lastName: "Estrada",
+  isLoggedIn: false,
+  cars: [
+    {
+      brand: "Toyota",
+      color: "Blue",
+    },
+    {
+      brand: "Nissan",
+      color: "Red",
+    },
+  ],
+};
+
+export const UserContext = createContext<IUserContext>(initialState);
+
+export const useUserContext = () => {
+  const context = useContext(UserContext);
+  if (!context) {
+    throw new Error("You can't use the themecontext outside of the provider");
+  }
+  return context;
+};
+
+enum ActionKind {
+  Add = "ADD_TO_CAR",
+  Update = "UPDATE_THE_CAR",
+  Remove = "REMOVE_TO_CART",
+}
+
+type ActionCar = {
+  type: ActionKind;
+  payload: { newData?: ICar; oldData?: ICar };
+};
+
+const reducer = (state: IUserContext, action: ActionCar) => {
+  const { type, payload } = action;
+
+  switch (type) {
+    case ActionKind.Add:
+      if (payload.newData) {
+        state.cars.push(payload.newData);
+      }
+      return state;
```

Comment:
You should always avoid mutating state. instead you can return a new copy by spreading it
```suggestion
      return {
        ...state,
        cars: [
           ...state.cars,
           payload
          ]
      }
```